modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/chart/bar/CustomizedAxisTick.es.js

```diff
@@ -17,10 +17,20 @@ import {Text} from 'recharts';
 
 import ellipsize from '../../../utils/ellipsize.es';
 
-export default ({payload, x, y}) => (
-	<Text textAnchor="middle" verticalAnchor="start" width={180} x={x} y={y}>
-		{payload.value.length > 34
-			? ellipsize(payload.value, 34)
+const MAX_SIZE = 34;
+
+const minimize = (size) => (size > 5 ? MAX_SIZE / 2 : MAX_SIZE);
+
+export default ({payload, size, x, y}) => (
+	<Text
+		textAnchor="middle"
+		verticalAnchor="start"
+		width={220 - size * 17}
```

Comment:
@alinedoleron, what these magic numbers `220` and `17` mean? Can we make them constants as `MAX_SIZE`?

Reply:
I'll refactor the code